messaging/relayer/internal/msgobserver/eth/observer/observer_impl_v1.go

```diff
@@ -56,147 +67,87 @@ func (o *ObserverImplV1) Start() error {
 		if err != nil {
 			return err
 		}
-		o.stopsMap = make(map[string]*map[string]chan bool)
-		observers, err := o.ListObserves()
+		exists, err := o.ds.Has(context.Background(), datastore.NewKey(activeKey))
 		if err != nil {
 			o.ds.Close()
 			return err
 		}
-		for chainID, addrs := range observers {
-			for _, addr := range addrs {
-				id, _ := big.NewInt(0).SetString(chainID, 10)
-				ap, err := o.ds.Get(context.Background(), dsKey(id, addr))
-				if err != nil {
-					o.ds.Close()
-					return err
-				}
-				chain, err := ethclient.Dial(string(ap))
-				if err != nil {
-					o.ds.Close()
-					return err
-				}
-				end := make(chan bool, 1)
-				go o.routine(id, chain, addr, end)
-
-				stops, ok := o.stopsMap[id.String()]
-				if !ok {
-					stopsNew := make(map[string]chan bool)
-					stops = &stopsNew
-					o.stopsMap[id.String()] = stops
-				}
-				(*stops)[addr.String()] = end
+		if exists {
+			data, err := o.ds.Get(context.Background(), datastore.NewKey(activeKey))
+			if err != nil {
+				o.ds.Close()
```

Comment:
nit:
is it possible to minimise the number of `o.ds.Close()` calls in this function. It adds a bit of noise that distracts from the core behaviour of the fn. perhaps a defer at the start of the function like so:
```
func (o *ObserverImplV1) Start() (err error) {
 defer func() {
   if err != nil && o.ds != nil {
     o.ds.Close()
   }
}()
```

Reply:
sounds good